### PR TITLE
estimate private tx bacthes

### PR DESCRIPTION
**Pull request by Kevingislason**
## Description

Estimates number of transaction batches required when estimating fees

---

Before we can merge this PR, please make sure that all the following items have been checked off. If any of the checklist items are not applicable, please leave them but write a little note why.

- [x] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work
- [x] Re-reviewed files changed in the Github PR explorer

## If PR to `staging`
- [x] Updated relevant documentation in the code
- [x] Test code changes manually, and describe your procedure in a comment on this PR
- [x] Mark this PR with the correct milestone

### CODE REVIEW

File: src/pages/SendPage/SendContext/index.tsx [hunk 1]
```diff
@@ -647,67 +650,137 @@ export const SendContextProvider = (props) => {
...
+      } else if (isPrivateTransfer()) {
+        tx = api.tx(dummyPrivateTransfer);
+      } else if (isToPublic()) {
+        tx = api.tx(dummyToPublic);
+      }
+      return tx;
+    };
+
+    const getFeeEstimate = async () => {
+      console.log('getFeeEstimate');
```
Comment (jasonintju) [inline src/pages/SendPage/SendContext/index.tsx:747]: a console here

**Review by jasonintju — DISMISSED** [review]

File: src/contexts/mantaWalletContext.tsx [hunk 2]
```diff
@@ -435,6 +433,39 @@ export const MantaWalletContextProvider = ({
...
+        const transferPostsCount = await privateWallet?.estimateTransferPostsCount(
+          {
+            network,
+            transactionType,
+            assetId: targetBalance.assetType.assetId.toString(),
+            amount: targetBalance.valueAtomicUnits.toString(),
+            zkAddressOrPolkadotAddress: address,
+          }
+        );
+        return Math.ceil(transferPostsCount / 2);
```
Comment (jasonintju) [inline src/contexts/mantaWalletContext.tsx:462]: Here, I think the gas fee you mocked includes `two transferPosts`. So how to mock the gas fee that only includes `one transferPost`?

**Comment by jasonintju** [comment]
Btw, the `transferPostsCount` returned from the api now are wrong. We have to wait for runtime check.

File: src/contexts/mantaWalletContext.tsx (hunk 2 above)
Comment (Kevingislason) [reply src/contexts/mantaWalletContext.tsx:463]: I think all transactions contain two transfer posts, but I'm not 100% sure

File: src/constants/DummyTransactions.ts [hunk 3]
```diff
@@ -1,3 +1,8 @@
+
```
[the remark this answers is not in the document]
Comment (Kevingislason) [reply src/constants/DummyTransactions.ts:1]: I added a log for each transaction type to log a serialized transaction. Then I did some arbitrary transactions locally, looked at the console, and copy-pasted those serialized transactions. The size should be the same for all transactions without regard to parameters, unless the number of transfer posts per batch are variable ( I don't think they are).

**Review by jasonintju — APPROVED** [review]

**Pull request merged**